packages/types/src/config.ts

```diff
@@ -45,3 +45,12 @@ export interface SettingsOptions {
   testingType?: 'component' |'e2e'
   args?: AllModeOptions
 }
+
+export type BannerState = {
+  lastShown?: number | null
+  dismissed?: number | null
+}
+
+export type BannersState = {
+  [bannerId: string]: BannerState | null
```

Comment:
`bannerId` is always known, right? I wonder if we can type this - it'll be very hard to understand what banners are available down the track.

If you make `BannerIds` like this

```ts
export const BannerIds = {
  ACI_082022_LOGIN: 'aci_082022_login',
  ACI_082022_CREATE_ORG: 'aci_082022_createOrganization',
  ACI_082022_CONNECT_PROJECT: 'aci_082022_connectProject',
  ACI_082022_RECORD: 'aci_082022_record',
} as const
```

(note `as const`) I think we should be able to type this more strongly (not just `string`).

Bit of a nitpick but having things typed properly will probably help when we have a lot more banners in the future and we've all forgotten the different types that exist, etc.

Reply:
🤔 Good callout. I probably can't directly use the referenced `BannerIds` struct here since that would add a dependency on `@cypress/app` to the `@cypress/types` package, but I'll see about adding a new `BannerId` union type that can be used both here and on the struct

Reply:
Hmm should we just move the `BannerIds` to `@packages/types`? I know it feels kind of bad to just dump whatever we need in `packages/types` but I don't find it so bad once you get over the fact it's sort of a melting pot for anything and everything. Then `packages/app` can depend on `packages/types` - not a big deal.

Reply:
I like it - I hadn't considered moving the `const` over to the typings. Done in [`da3c2e3`](https://github.com/cypress-io/cypress/pull/23256/commits/da3c2e3be43d84ccb4816bc2a08b341c394462d3)